src/main/java/com/clearspring/analytics/stream/cardinality/HyperLogLogPlus.java

```diff
@@ -789,11 +805,11 @@ public void addAll(HyperLogLogPlus other) throws HyperLogLogPlusMergeException {
 
         if ((format == Format.SPARSE) && (other.format == Format.SPARSE)) {
             sparseSet = mergeEstimators(other);
-            // Convert to normal mode if needed. 
-            // Since offer trigger the switch to the normal mode on only when 
-            // the tmpSet is full and the threshold is reached, we follow the same 
+            // Convert to normal mode if needed.
+            // Since offer trigger the switch to the normal mode on only when
+            // the tmpSet is full and the threshold is reached, we follow the same
             // behavior here to ease testing
-            if (sparseSet.length > (sparseSetThreshold + sortThreshold)) {
+            if (sparseSet.length > (sparseSetThreshold * (SPARSE_SET_TEMP_SET_RATIO + 1)) / SPARSE_SET_TEMP_SET_RATIO) {
```

Comment:
I think the comments above the previous logic were wrong, and that guard was imprecise. I went ahead and fixed the base version since the case reflects the possibility for hllps to sometimes use more memory in sparse mode than they would in normal mode (for an indefinite amount of time), and that isn't super cool.